### PR TITLE
ioctl: add frameintervals

### PR DESCRIPTION
**Pull request by aesteve-rh**
Add support for VIDIOC_ENUM_FRAMEINTERVALS ioctl.

### CODE REVIEW

File: lib/src/ioctl/frameintervals.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+use std::os::unix::io::AsRawFd;
+use thiserror::Error;
+
+use crate::{bindings, PixelFormat};
+
+pub trait FrameSize {
```
Comment (Gnurou) [inline lib/src/ioctl/frameintervals.rs:6]: Shouldn't this be called `FrameIntervals`?

File: lib/src/ioctl/frameintervals.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+#[doc(hidden)]
+mod ioctl {
+    use crate::bindings::v4l2_frmivalenum;
+    nix::ioctl_readwrite!(vidioc_enum_frameintervals, b'V', 75, v4l2_frmivalenum);
+}
+
+#[derive(Debug, Error)]
+pub enum FrameSizeError {
```
Comment (Gnurou) [inline lib/src/ioctl/frameintervals.rs:23]: Same here, I'd expect `FrameIntervalError` but maybe I am missing the intended logic.

Comment (aesteve-rh) [reply lib/src/ioctl/frameintervals.rs:23]: No, you are right... a copy&paste problem, thanks for checking!

**Comment by Gnurou** [comment]
Looking good, thanks!

**Pull request merged**